test/test_security_constrained_models.jl

```diff
@@ -0,0 +1,384 @@
+        #Test Expected error since no SCUC valid attributes were added
+        @test build!(
+            ps_model;
+            console_level = Logging.AboveMaxLevel,  # Ignore expected errors.
+            output_dir = mktempdir(; cleanup = true),
+        ) == PSI.ModelBuildStatus.FAILED
+
+        for branch_name in lines_outages[sys]
+            transition_data = GeometricDistributionForcedOutage(;
+                mean_time_to_recovery = 10,
+                outage_transition_probability = 0.9999,
+            )
+            component = get_component(ACBranch, sys, branch_name)
+            add_supplemental_attribute!(sys, component, transition_data)
+        end
+
+        @test build!(ps_model; output_dir = mktempdir(; cleanup = true)) ==
+              PSI.ModelBuildStatus.BUILT
+        psi_constraint_test(ps_model, constraint_keys)
+
+        moi_tests(
+            ps_model,
+            test_results[sys][1],
+            test_results[sys][2],
+            test_results[sys][3],
+            test_results[sys][4],
+            test_results[sys][5],
+            false,
+        )
+        psi_checkobjfun_test(ps_model, objfuncs[ix])
+        psi_checksolve_test(
+            ps_model,
+            [MOI.OPTIMAL, MOI.ALMOST_OPTIMAL],
+            test_obj_values[sys],
+            10000,
+        )
+    end
+end
+
+@testset "Security Constrained Network DC-PF with PTDF/LODF Model using Rating B for Post-Contingency Flows, dynamic line ratings and outages that should be neglected" begin
+    normal_op_dlr_factors = vcat([fill(x, 6) for x in [1.15, 1.05, 1.1, 1.0]]...)
+    postcontingency_dlr_factors = vcat([fill(x, 6) for x in [1.25, 1.15, 1.2, 1.1]]...)
+    dlr_dict = Dict(
+        "dynamic_line_ratings" => normal_op_dlr_factors,
+        "Post_contingency_dynamic_line_ratings" => postcontingency_dlr_factors,
+    )
+    line_device_model = DeviceModel(
+        Line,
+        StaticBranch;
+        time_series_names = Dict(
+            DynamicBranchRatingTimeSeriesParameter => collect(keys(dlr_dict))[1],
+            PostContingencyDynamicBranchRatingTimeSeriesParameter =>
+                collect(keys(dlr_dict))[2],
+        ))
+    TapTransf_device_model = DeviceModel(
+        TapTransformer,
+        StaticBranch;
+        time_series_names = Dict(
+            DynamicBranchRatingTimeSeriesParameter => collect(keys(dlr_dict))[1],
+            PostContingencyDynamicBranchRatingTimeSeriesParameter =>
+                collect(keys(dlr_dict))[2],
+        ))
+    template = get_thermal_dispatch_template_network(SecurityConstrainedPTDFPowerModel)
+    c_sys5 = PSB.build_system(PSITestSystems, "c_sys5")
+    c_sys14 = PSB.build_system(PSITestSystems, "c_sys14")
+    c_sys14_dc = PSB.build_system(PSITestSystems, "c_sys14_dc")
+    systems = [c_sys5, c_sys14, c_sys14_dc]
+    objfuncs = [GAEVF, GQEVF, GQEVF]
+    constraint_keys = [
+        PSI.ConstraintKey(RateLimitConstraint, PSY.Line, "lb"),
```

Comment:
Refactor to remove very large code duplication

Reply:
@SebastianManriqueM